TestVaultTokenRenewal is flaky

The integration test `TestVaultTokenRenewal` is flaky. For example in [this CI run](https://github.com/grafana/mimir/actions/runs/7917849168/job/21615241928?pr=7388) we've got:

```
2024-02-15T15:13:08.1919903Z === RUN   TestVaultTokenRenewal

[... REDACTED ...]

2024-02-15T15:13:08.2242443Z     vault_test.go:81: 
2024-02-15T15:13:08.2242880Z         	Error Trace:	/home/runner/work/mimir/mimir/integration/vault_test.go:81
2024-02-15T15:13:08.2243098Z         	Error:      	Received unexpected error:
2024-02-15T15:13:08.2244025Z         	            	unable to find metrics [cortex_vault_auth_success_total] with expected values. Last error: <nil>. Last values: [10]
2024-02-15T15:13:08.2244222Z         	Test:       	TestVaultTokenRenewal
2024-02-15T15:13:08.2244340Z 15:13:07 Killing mimir-1
2024-02-15T15:13:08.2244442Z 15:13:07 Killing vault
2024-02-15T15:13:08.2244575Z 15:13:07 Killing minio-9000
2024-02-15T15:13:08.2244671Z 15:13:07 Killing consul
2024-02-15T15:13:08.2244806Z 15:13:07 Killed minio-9000
2024-02-15T15:13:08.2244902Z 15:13:07 Killed vault
2024-02-15T15:13:08.2244997Z 15:13:07 Killed consul
2024-02-15T15:13:08.2245110Z 15:13:07 Killed mimir-1
2024-02-15T15:13:08.2245389Z --- FAIL: TestVaultTokenRenewal (75.96s)
```

--- a/integration/vault_test.go
+++ b/integration/vault_test.go
@@ -78,5 +78,5 @@ func TestVaultTokenRenewal(t *testing.T) {
 	// Check that the token lease has been updated before hitting max_ttl
 	require.NoError(t, mimir.WaitSumMetrics(e2e.GreaterOrEqual(2), "cortex_vault_token_lease_renewal_success_total"))
 	// Check that re-authentication occurred
-	require.NoError(t, mimir.WaitSumMetrics(e2e.Equals(2), "cortex_vault_auth_success_total"))
+	require.NoError(t, mimir.WaitSumMetrics(e2e.GreaterOrEqual(2), "cortex_vault_auth_success_total"))
 }

